### PR TITLE
EUI-4329-Several events for Divorce Bulk Cases fail case data validation via the XUI when all is ok via CCD data validation

### PR DESCRIPTION
**Pull request by leepavlouhmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4329

### Change description ###
added an additional formcontrol to the linkedCases formgroup to help identify when the abstractControl is of type linkedCases, then added an if statement further down the chain that correctly sets the value of 'CaseReference'

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/shared/components/palette/base-field/abstract-form-field.component.ts [hunk 1]
```diff
@@ -65,10 +65,14 @@ export abstract class AbstractFormFieldComponent {
...
-      parent.setControl(this.caseField.id, control);
+      if (control['controls'] && control['controls']['Id'] && control['controls']['Id'].value === 'LinkedCase') {
+        parent.setControl('CaseReference', control['controls']['CaseReference']);
+      } else {
+        parent.setControl(this.caseField.id, control);
+      }
     } else {
       parent.addControl(this.caseField.id, control);
-    }
+    };
```
Comment (adnanakgun) [inline src/shared/components/palette/base-field/abstract-form-field.component.ts:75]: Semi colon after a closing if clause? It is probably not breaking anything but let's not do that as it isn't standard practise.

**Review by adnanakgun — CHANGES_REQUESTED** [review]
One minor change requested.

File: src/shared/components/palette/datetime-picker/datetime-picker.component.ts [hunk 2]
```diff
@@ -127,7 +127,7 @@ export class DatetimePickerComponent extends AbstractFormFieldComponent implemen
       this.hideTime = true;
       this.checkTime = false;
       this.dateTimeEntryFormat = this.formatTranslationService.removeTime(this.dateTimeEntryFormat);
-      this.momentFormat = 'YYYY-MM-DD';
+      this.momentFormat = 'YYYY-MM-DD'
```
Comment (kuda1992) [inline src/shared/components/palette/datetime-picker/datetime-picker.component.ts:130]: Removed the semicolon why?

**Review by kuda1992 — CHANGES_REQUESTED** [review]
I don't see any new tests updated for component `write-case-link-field.component.ts` to test the new logic.

File: src/shared/components/palette/base-field/abstract-form-field.component.ts (hunk 1 above)
Comment (leepavlouhmcts) [reply src/shared/components/palette/base-field/abstract-form-field.component.ts:75]: i will remove this now.

File: src/shared/components/palette/datetime-picker/datetime-picker.component.ts (hunk 2 above)
Comment (leepavlouhmcts) [reply src/shared/components/palette/datetime-picker/datetime-picker.component.ts:130]: that was a mistake i think with the merge. will put the semi colon back

**Comment by leepavlouhmcts** [comment]
> I don't see any new tests updated for component `write-case-link-field.component.ts` to test the new logic.

added.

**Review by kuda1992 — DISMISSED** [review]
Thanks for adding the tests @leepavlouhmcts . Approved 😊

**Review by adnanakgun — DISMISSED** [review]

**Pull request merged**